src/utils/client.coffee

```diff
@@ -0,0 +1,59 @@
+getToken = (callback)->
+    receiveToken = (event) ->
+        window.removeEventListener 'message', receiveToken
+        {appName, token}  = event.data
+        callback? null, {appName, token}
+        callback = null
+    
+    unless window.parent?
+        return callback new Error('no parent window')
+    
+    unless window.parent?.postMessage
+        return callback new Error('get a real browser')
+    
+    window.addEventListener 'message', receiveToken, false
+    window.parent.postMessage { action: 'getToken' }, '*'
```

Comment:
Why did we choose to use `postMessage` for the token instead of putting it in the URL of the iframe?